engine/api/auth/ldapclient.go

```diff
@@ -225,6 +225,28 @@ func (c *LDAPClient) Bind(username, password string) error {
 	return nil
 }
 
+// BindDN
+func (c *LDAPClient) BindDN(dn, password string) error {
+  log.Debug("LDAP> Bind DN %s", dn)
+
+  if err := c.conn.Bind(dn, password); err != nil {
+    if shoudRetry(err) {
+      err = c.openLDAP(c.conf)
+      if err != nil {
+        return err
+      }
+      err = c.conn.Bind(dn, password)
+      if err != nil {
+        return err
+      }
+    } else {
```

Comment:
We can do early return: if !shouldretry

Reply:
Yeah, I just tried to keep the same structure of the other function  func (c *LDAPClient) Bind(username, password..